src/Platform/Microsoft.Testing.Platform.MSBuild/Tasks/InvokeTestingPlatformTask.cs

```diff
@@ -202,6 +242,32 @@ protected override string ToolName
     private bool IsCurrentProcessArchitectureCompatible() =>
         _currentProcessArchitecture == EnumPolyfill.Parse<Architecture>(TestArchitecture.ItemSpec, ignoreCase: true);
 
+    private string? TryGetRunCommand()
+    {
+        if (!RuntimeInformation.IsOSPlatform(OSPlatform.Windows))
+        {
+            // Currently fails on Linux and macOS.
+            // Not yet investigated.
+            return null;
+        }
+
+        // This condition specifically handles this part:
+        // https://github.com/dotnet/sdk/blob/5846d648f2280b54a54e481f55de4d9eea0e6a0e/src/Tasks/Microsoft.NET.Build.Tasks/targets/Microsoft.NET.Sdk.targets#L1152-L1155
+        // The more correct logic is implementing https://github.com/microsoft/testfx/issues/5091
+        if (IsNetCoreApp &&
```

Comment:
Please add the comment about the reason that we saw in the app, where it looked at the current process to figure out the path, and then it failed to find a resource. 

Because when running exe we were in the current folder where the resource is, but if we run dotnet.exe exec theapp.dll it thinks we are in program files/dotnet and crashes.

Reply:
Done